src/OracleService/Protocols/OracleHttpsProtocol.cs

```diff
@@ -82,7 +83,29 @@ public async Task<(OracleResponseCode, string)> ProcessAsync(Uri uri, Cancellati
                 return (OracleResponseCode.Error, message.StatusCode.ToString());
             if (!Settings.Default.AllowedContentTypes.Contains(message.Content.Headers.ContentType.MediaType))
                 return (OracleResponseCode.ContentTypeNotSupported, null);
-            return (OracleResponseCode.Success, await message.Content.ReadAsStringAsync(cancellation));
+            if (message.Content.Headers.ContentLength.HasValue && message.Content.Headers.ContentLength > OracleResponse.MaxResultSize)
+                return (OracleResponseCode.ResponseTooLarge, null);
+
+            byte[] buffer = new byte[OracleResponse.MaxResultSize + 1];
+            var stream = message.Content.ReadAsStream(cancellation);
+            var read = await stream.ReadAsync(buffer, 0, buffer.Length, cancellation);
+
+            if (read > OracleResponse.MaxResultSize)
+                return (OracleResponseCode.ResponseTooLarge, null);
+
+            var encoding = GetEncoding(message.Content.Headers);
+            return (OracleResponseCode.Success, encoding.GetString(buffer, 0, read));
+        }
+
+        private static Encoding GetEncoding(HttpContentHeaders headers)
+        {
+            Encoding encoding = null;
+            if ((headers.ContentType != null) && (headers.ContentType.CharSet != null))
+            {
+                encoding = Encoding.GetEncoding(headers.ContentType.CharSet);
+            }
+
+            return encoding ?? Encoding.UTF8;
```

Comment:
Not sure, but shouldn't we *always* enforce strict UTF8 as it is done for OracleNeoFSProtocol?
https://github.com/neo-project/neo-modules/blob/b639e84c8fa7aa33c063ba764b27b9f93ef0ef04/src/OracleService/Protocols/OracleNeoFSProtocol.cs#L100

Reply:
@roman-khimov, what do you think? In neo-go it's always utf-8: https://github.com/nspcc-dev/neo-go/blob/aca12b58c00410d765f205b752afdc716dd9fade/pkg/services/oracle/response.go#L76

Reply:
Before it worked like this, I think that it's better

Reply:
I don't think it ever worked with non-UTF-8:
https://github.com/neo-project/neo-modules/blob/b639e84c8fa7aa33c063ba764b27b9f93ef0ef04/src/OracleService/OracleService.cs#L508-L513
`JToken.Parse` also insists on UTF-8.

Reply:
Also, it would be nice to make it symmetrical with OracleNeoFSProtocol (it always enforses UTF-8).

Reply:
What is the error that do you think that we should throw in that case?

Reply:
We should probably return just an `OracleResponseCode.Error` (0xff) in this case, because `OracleResponseCode.ContentTypeNotSupported` is returned when response content type doesn't match the configured one.

Reply:
Resolved.